classes/Builder/Payload/OrderPayloadBuilder.php

```diff
@@ -48,38 +48,51 @@ public function __construct(array $cart)
 
     /**
      * Build payload with cart details
+     *
+     * @param bool $expressCheckout Allow to build the payload with more or less content depending if
+     * the customer use expresse checkout or not
      */
-    public function buildFullPayload()
+    public function buildFullPayload($expressCheckout = false)
     {
         parent::buildFullPayload();
 
         $this->buildBaseNode();
         $this->buildAmountBreakdownNode();
         $this->buildItemsNode();
+
+        if (!$expressCheckout) {
+            $this->buildShippingNode();
+            $this->buildPayerNode();
+        }
+
+        $this->buildApplicationContextNode($expressCheckout);
     }
 
     /**
      * Build payload without cart details
+     *
+     * @param bool $expressCheckout Allow to build the payload with more or less content depending if
+     * the customer use expresse checkout or not
      */
-    public function buildMinimalPayload()
+    public function buildMinimalPayload($expressCheckout = false)
```

Comment:
You're adding a prameter for a new feature. What will happen when we add, let's say ... 4 more features? We will get our payloads with `$buider->buildMinimalPayload(true, false, false, true)`?

This will be difficult to understand. My advice would be to set these options via setters, in case you absolutely need them everywhere.

For instance:

```php
$builder->toggleExpressCheckout(true);
$builder->buildMinimalPayload();
```

What do you think?

Reply:
Done! :)